Add PHPUnit test for _mce_set_direction function

pbearne opened the pull request:
This commit introduces a new PHPUnit test for the _mce_set_direction function within WordPress. The test checks the functionality of the text direction setting, ensuring it correctly switches between 'rtl' and 'ltr' options. Furthermore, the changes improve code traceability by linking the test to ticket 60219.


Trac ticket: https://core.trac.wordpress.org/ticket/60219

pbearne committed:
Add PHPUnit test for _mce_set_direction function

This commit introduces a new PHPUnit test for the _mce_set_direction function within WordPress. The test checks the functionality of the text direction setting, ensuring it correctly switches between 'rtl' and 'ltr' options. Furthermore, the changes improve code traceability by linking the test to ticket 60219.

jdstphen00 commented:
PHPUnit test for the _mce_set_direction function. However, please note that I don't have the actual implementation of the _mce_set_direction function, so I'll create a generic example for demonstration purposes. You'll need to adapt it to match the actual implementation of your function.

Let's assume _mce_set_direction is a function that takes a string and a direction (e.g., 'ltr' or 'rtl') and sets the text direction in some way. Here's a PHPUnit test for it:
`<?php

class YourClassTest extends PHPUnit\Framework\TestCase {

    // Import the class containing _mce_set_direction function
    // use YourClass;

    public function testSetDirectionLtr()
    {
        // Arrange
        $inputString = 'Hello, World!';
        $direction = 'ltr';

        // Act
        $result = YourClass::_mce_set_direction($inputString, $direction);

        // Assert
        $this->assertEquals('Hello, World!', $result);
        // Add more assertions based on the actual behavior of _mce_set_direction
    }

    public function testSetDirectionRtl()
    {
        // Arrange
        $inputString = 'مرحبا بك في العالم';
        $direction = 'rtl';

        // Act
        $result = YourClass::_mce_set_direction($inputString, $direction);

        // Assert
        // Add assertions based on the expected result for RTL direction
    }

    // Add more test cases as needed

}

> 

Replace YourClass with the actual class containing the _mce_set_direction function. In each test method, you set up the initial conditions (Arrange), perform the action (Act), and then assert that the result is as expected (Assert). Adjust the input values and assertions based on the actual behavior of your _mce_set_direction function.

Make sure to include the necessary dependencies and adjust the class and method names according to your actual code structure.


HAVE A LOOK ON FOLLOWING SITES AS A SAMPLE 
https://www.webmastershall.com
https://www.techuy.com
https://www.edulize.com
https://www.travelguidebook.org
https://www.nogentech.org
https://stuffablog.com
https://www.bravotech.org
https://tricksmode.com

SergeyBiryukov reviewed (changes requested):
Thanks for the PR!

I think we can save the original text direction to a variable and then restore it before performing the assertion, like this:
```
$actual = _mce_set_direction( $mce_init );
$this->assertSameSets( $mce_init, $actual );

$orig_text_dir             = $wp_locale->text_direction;
$wp_locale->text_direction = 'rtl';
$actual = _mce_set_direction( $mce_init );
$wp_locale->text_direction = $orig_text_dir;

$this->assertSameSets( $expected, $actual );
```
This way, if the last assertion fails, it won't affect any other tests because the original text direction value is already restored.

pbearne committed:
Update _mce_set_direction function test in PHPUnit

This update improves the test for _mce_set_direction function by storing the actual result in a variable before carrying out the tests. Moreover, it safeguards the original text direction by storing it in a variable to be restored after the test, ensuring consistent and reliable test results.

swissspidy commented:
https://core.trac.wordpress.org/changeset/57718
